include/aero/interpolator/interpolator.h

```diff
@@ -0,0 +1,40 @@
+#ifndef AERO_INTERPOLATOR_H
+#define AERO_INTERPOLATOR_H
+
+#include <aero/aero.h>
+#include <aero/array/array.h>
+#include <aero/grid/grid.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// This opaque type represents an implementation of an interpolator
+typedef struct aero_interpolator_data_t aero_interpolator_data_t;
+
+/// Behaviors and data associated with an Interpolator. All of these
+/// functions are passed the data pointer created by an Interplator
+/// constructor.
+typedef struct aero_interpolator_t aero_interpolator_t;
+typedef struct aero_interpolator_t {
+  /// pointer to the interpolator data
+  aero_interpolator_data_t *data_;
+  /// returns a clone (deep-copy) of an interpolator
```

Comment:
I'm not sure we need this (in the same sense as we don't need the C++ copy constructor). Can we go without for now and reintroduce if we find a need for it?

Reply:
not quite sure what you're referring to here - the `clone()` function?

Reply:
Yes, that's the one. Sorry for not being clear!

Reply:
no problem, just wanted to make sure I remove the right thing. should `clone()` be removed from the C++ and Fortran implementations as well?

Reply:
Yes. Maybe let's just comment them out for now and reintroduce if needed?

Reply:
I don't mind just deleting them - we can always grab them from the git history as long as we don't merge & squash